### PR TITLE
Add ignoreFontFamilyName option to font-family-no-duplicate

### PR DESCRIPTION
**Pull request by CAYdenberg**
> Which issue, if any, is this issue related to?

Issue #2086 

> Is there anything in the PR that needs further explanation?

I think it's pretty self-explanatory but this is my first PR to a major project so let me know if anything needs changing.

### CODE REVIEW

File: lib/rules/font-family-no-duplicate-names/README.md [hunk 1]
```diff
@@ -45,3 +45,21 @@ a { font: 1em "Arial", "sans-serif", sans-serif; }
 ```css
 a { font: normal 14px/32px -apple-system, BlinkMacSystemFont, sans-serif; }
 ```
+
+## Optional secondary options
+
+### `ignoreFontFamilyNames: ["monospace"]`
```
Comment (m-allanson) [inline lib/rules/font-family-no-duplicate-names/README.md]: Can you change this to show the types of values that can be accepted here?
```markdown
### `ignoreFontFamilyNames: ["/regex/", "string"]`
```

File: lib/rules/font-family-no-duplicate-names/README.md [hunk 2]
```diff
@@ -45,3 +45,21 @@ a { font: 1em "Arial", "sans-serif", sans-serif; }
...
+
+Given:
+
+```js
+["monospace"]
```
Comment (m-allanson) [inline lib/rules/font-family-no-duplicate-names/README.md]: I think it'd be good to include a regex example here too.

**Comment by CAYdenberg** [comment]
Good suggestions. You got me all confused about case-sensitivity so I added another test as well.

File: lib/rules/font-family-no-duplicate-names/README.md (hunk 1 above)
Comment (CAYdenberg) [inline lib/rules/font-family-no-duplicate-names/README.md]: Addressed in ee5279e

File: lib/rules/font-family-no-duplicate-names/README.md (hunk 2 above)
Comment (CAYdenberg) [inline lib/rules/font-family-no-duplicate-names/README.md]: Addressed in ee5279e

**Comment by m-allanson** [comment]
@CAYdenberg 👍 for extra tests. 

I should have explained myself more clearly for the docs changes in the previous comments. 

The [`ignoreFontFamilyNames` title](https://github.com/stylelint/stylelint/pull/2314/files?diff=split#diff-5c425a57dfd87ced29b7b6a300031f4aR51) should show the _types_ of values it accepts - regexes or strings e.g. `["/regex/", "string"]`

Then the [example usage](https://github.com/stylelint/stylelint/pull/2314/files?diff=split#diff-5c425a57dfd87ced29b7b6a300031f4aR56) should show an _example_ regex and an _example_ string. e.g. `["/^my-/", "custom"]`

You can use the [ignoreAtRules option of at-rule-no-unknown](https://github.com/stylelint/stylelint/blob/master/lib/rules/at-rule-no-unknown/README.md#ignoreatrules-regex-string) as a reference.

Can you update those parts of the readme along with the regex code example [on L66](https://github.com/stylelint/stylelint/pull/2314/files?diff=split#diff-5c425a57dfd87ced29b7b6a300031f4aR66)? This might all seem overly fussy but there are a lot of rules in stylelint so we aim for strong consistency across the docs. 

Other than the docs update this LGTM, thanks again for working on it.

**Comment by CAYdenberg** [comment]
+1 for fussiness - I don't think someone who isn't fussy would be using/contributing to/writing a CSS linter!

Is the issue that the example `("/Regex/")` doesn't look enough like a regular expression? I mean, it _could_ be, but I guess I see your point. Most font family names are capitalised and have spaces (not hyphens) so I guess my thought was there would be some value in providing an example that looks like something someone might actually use. Would ("/^Regex/") be more appropriate? Or would you prefer `("/^my-/") since it's closer to the other examples.

I do see what you mean about the title - I just outright missed that.

**Comment by m-allanson** [comment]
>Is the issue that the example ("/Regex/") doesn't look enough like a regular expression?

Exactly - I think it's confusing if the example regex includes the string "Regex" in it. I see your point about font family names though. We could use a more realistic example regex - `"/^My Font /"` or `"/^CustomFamily/"`?

So we'd have `["/regex/", "string"]` in the title, and something like `["/^My Font /", "custom"]` in the example.

**Comment by CAYdenberg** [comment]
Got it. That should do it? I left `monospace` as the example string since that will probably cover most use cases for this option. Other than that it's exactly as you prescribed.

**Comment by m-allanson** [comment]
That LGTM, thanks again!

**Comment by davidtheclark** [comment]
Looks great. Thanks @CAYdenberg! I notice you're doing a lot to help out!

**Comment by davidtheclark** [comment]
Added to changelog:

> -   Added: `ignoreFontFamilyName` option to `font-family-no-duplicate` ([#2314](https://github.com/stylelint/stylelint/pull/2314)).

**Comment by webdesignberlin** [comment]
Any Example there, how i can use the ignoreFontFamilyName Rule within the stylelintrc?

I have try this:

``
"font-family-no-duplicate-names": [
     true, { "ignore-font-family-name": ["monospace"] }
],
``

But i became an error:
``
Invalid option name "ignore-font-family-name" for rule "font-family-no-duplicate-names"
``

Version:
`
$ ./node_modules/.bin/stylelint -v
8.0.0
`

**Comment by evilebottnawi** [comment]
@webdesignberlin use `ignoreFontFamilyName` instead `ignore-font-family-name`

**Comment by webdesignberlin** [comment]
@evilebottnawi 

thx, but ai tried this also. same error:
`
Invalid Option: Invalid option name "ignoreFontFamilyName" for rule "font-family-no-duplicate-names"
`

**Comment by webdesignberlin** [comment]
Oje, he, she, it the s goes with :) ignoreFontFamilyNames works.

via Source Code: 
https://github.com/stylelint/stylelint/blob/74d9d05f25e8b865bc21801ec5759c4229592f71/lib/rules/font-family-no-duplicate-names/index.js#L46

**Comment by jeddy3** [comment]
For future visitors... the `ignoreFontFamilyNames` option is documented [here](https://github.com/stylelint/stylelint/blob/master/lib/rules/font-family-no-duplicate-names/README.md#ignorefontfamilynames-regex-string).

**Pull request merged**